### PR TITLE
Solution Explorer showing netcoreapp5.0-windowsNT file first in case of multiple frameworks

### PR DESCRIPTION
**Pull request by Anipik**
VS loads file for the first Target Framework in the TargetFrameworks list.
So by defaut we will be loading files for netcoreapp5-windows_NT. 
We have an issue tracking to fix it more appropriately eg showing all the files or something, but it might require work from the vs team.

Related Issue https://github.com/dotnet/runtime/issues/31902

### CODE REVIEW

File: src/libraries/System.Security.Cryptography.X509Certificates/src/System.Security.Cryptography.X509Certificates.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <AssemblyName>System.Security.Cryptography.X509Certificates</AssemblyName>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <NoWarn>$(NoWarn);CS3016;CA5379;CA5384;CA5385</NoWarn>
-    <TargetFrameworks>$(NetCoreAppCurrent)-OSX;$(NetCoreAppCurrent)-Unix;$(NetCoreAppCurrent)-Windows_NT</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent)-OSX;$(NetCoreAppCurrent)-Windows_NT;$(NetCoreAppCurrent)-Unix</TargetFrameworks>
```
Comment (danmosemsft) [inline src/libraries/System.Security.Cryptography.X509Certificates/src/System.Security.Cryptography.X509Certificates.csproj]: Why OSX first? And below.

**Comment by danmosemsft** [comment]
We're picking Windows by default here simply because VS users are inevitably on Windows ?

**Review by danmoseley — APPROVED** [review]
Looks good assuming you consistently put the same target framework first

**Comment by Anipik** [comment]
>  We're picking Windows by default here simply because VS users are inevitably on Windows ? 

yes that is my assumption.

**Comment by Anipik** [comment]
> Does VS Code not use this?

Vscode just behave like an editor and show you all the files in the directory structure.

**Comment by Anipik** [comment]
Merging this as the failure is known and we have an issue tracking this https://github.com/dotnet/runtime/issues/31888

**Comment by ericstj** [comment]
I wonder if we can include non-compile items as none so that they still appear in SLN hierarchy.

**Comment by Anipik** [comment]
> I wonder if we can include non-compile items as none so that they still appear in SLN hierarchy.

i can give it a try.

**Comment by ericstj** [comment]
@dsplaisted or @nguerrera may have some experience on the efficacy of this.  I believe that is the pattern folks tend to use with SDK projects, but I could be mistaken.

**Comment by nguerrera** [comment]
We have tended to avoid conditional Compile items, and use #if to exclude things. But we don't have a lot of platform or target framework specific files. I would ask the project system team about the efficacy of other approaches like this one. cc @davkean @drewnoakes

**Comment by davkean** [comment]
Including as `<None>` is the approach the SDK used to give for free, but now you need to opt in: https://github.com/dotnet/sdk/issues/2337. Workaround is at the bottom of the issue.

**Comment by ericstj** [comment]
Nice, that should do it.  That'd only work for things under the library folder, not common.  If we wanted it to work for Common too we should probably change from using Compile with conditions and instead have some named items for other platforms, eg: `CompileWindows` `CompileLinux` etc.  Not sure I like that so much, we could do an outer/inner build to collect all the items across all frameworks, then emit that to an intermediate file that could later be imported by inner builds in VS (evil laugh).  What do you think about that?

**Comment by danmosemsft** [comment]
Our conditions are not all simple checks on the target platform so may not map well to specific item types. Maybe excluding common files that are specific to other platforms is not a big burden, so long as all files in the library proper are present and editable.

**Comment by ericstj** [comment]
Yeah, I don't think generalizing conditions to items is a good idea.

The workaround of globbing things under library gets us pretty close to acceptable.  I think if we cared about showing all the common source used by the lib we should do some outer/inner-build step to gather items for VS experience.

**Comment by danmosemsft** [comment]
I assume the goals are
1. Be able to see and edit the code. Sounds like there's a way to do that.
2. Navigation (FR, GTD) within the code. I assume this won't work because VS needs to see the files going to the compiler - unless it's OmniSharp style navigation that doesn't seem to need a build? 
3. Refactoring both configurations at once, and build once to see impact on both. I assume there's no reasonable way to make that work and that's not a big deal.

**Comment by ericstj** [comment]
I suspect 1&2 will work, since the editor should still be able to show context based on drop-down, but I can't be certain until we try it out.

3 : build both works today, refactoring existing files works today.  Renaming files won't work until some better support from project system (since we would be editing the none items, not compile items).

**Comment by davkean** [comment]
If you are using multi-targeting as designed, you will get context & IntelliSense for these files as we'll build those other configurations (via design-time build) and send them onto Roslyn.

Refactoring doesn't work correctly across multi-targeting context, that would be request on Roslyn.

**Comment by Anipik** [comment]
@davkean  would it be possible to choose the default TargetFramework dynamically through a running  target rather than always choosing the first one in the TargetFrameworks property ?

**Comment by davkean** [comment]
@Anipik We could consider adding a property to pick the default one (or UI gesture to change the "current one"), but running a target would not work. 

That would require loading a configuration, running a build over that configuration, grab the result of that build and use it to pick a different configuration. We show Solution Explorer contents today without even running a build.

**Comment by davkean** [comment]
To expand on my point, to evaluate (figure out all properties/items) and build (running a target) over a project you need to pick a "configuration".

In a normal project, that is a combination of `<Configuration>` and `<Platform>` dimensions, say `Debug|AnyCPU`. 

These properties are special and need to be known upfront, they can't come from a target because you have a chicken and egg problem; can't run a target until you have a configuration and can't get configuration because you can't run a target.

For multi-targeting projects, `TargetFramework` becomes a 3rd dimension, so now you need a combination of  `<TargetFramework>`, `<Configuration>` and `<Platform>`; `Debug|AnyCPU|net45`. This is why TargetFramework cannot come from a target.

**Comment by Anipik** [comment]
> @Anipik We could consider adding a property to pick the default one (or UI gesture to change the "current one"), 

Thanks @davkean for the explanation. It would be nice to have a ui gesture to change the targetFramework. do we need to create an issue or something for that feature work ?

**Comment by Anipik** [comment]
>  Renaming files won't work until some

@ericstj I tried renaming the files and it was working, or maybe I didn't understand the exact issue here.

**Pull request merged**